src/main/java/org/candlepin/resource/ContentResource.java

```diff
@@ -33,18 +40,53 @@ public class ContentResource implements ContentApi {
     private final ContentCurator contentCurator;
     private final I18n i18n;
     private final ModelTranslator modelTranslator;
+    private final Configuration config;
 
     @Inject
-    public ContentResource(ContentCurator contentCurator, I18n i18n, ModelTranslator modelTranslator) {
+    public ContentResource(ContentCurator contentCurator, I18n i18n, ModelTranslator modelTranslator,
+        Configuration config) {
         this.i18n = Objects.requireNonNull(i18n);
         this.contentCurator = Objects.requireNonNull(contentCurator);
         this.modelTranslator = Objects.requireNonNull(modelTranslator);
+        this.config = Objects.requireNonNull(config);
     }
 
     @Override
-    public Stream<ContentDTO> getContents() {
-        return this.contentCurator.listAll()
-            .stream()
+    public Stream<ContentDTO> getContents(Integer page, Integer perPage, String order, String sortBy) {
+        PageRequest pageRequest = ResteasyContext.getContextData(PageRequest.class);
+        ContentQueryArguments queryArgs = new ContentQueryArguments();
+        long count = this.contentCurator.getContentCount(queryArgs);
+
+        if (pageRequest != null) {
+            Page<Stream<ContentDTO>> pageResponse = new Page<>();
+            pageResponse.setPageRequest(pageRequest);
+
+            if (pageRequest.isPaging()) {
+                queryArgs.setOffset((pageRequest.getPage() - 1) * pageRequest.getPerPage())
+                    .setLimit(pageRequest.getPerPage());
+            }
+
+            if (pageRequest.getSortBy() != null) {
+                boolean reverse = pageRequest.getOrder() == PageRequest.Order.DESCENDING;
```

Comment:
Instead of explicitly setting the order as descending, lets use the default variable (in case we ever change the default): `PageRequest.DEFAULT_ORDER`